packages/go_router/lib/src/delegate.dart

```diff
@@ -66,11 +66,12 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     // Fallback to onExit if maybePop did not handle the pop
     final GoRoute lastRoute = currentConfiguration.last.route;
     if (lastRoute.onExit != null && navigatorKey.currentContext != null) {
-      return !(await lastRoute.onExit!(
-        navigatorKey.currentContext!,
-        currentConfiguration.last
-            .buildState(_configuration, currentConfiguration),
-      ));
+      return !(await lastRoute.onExit?.call(
```

Comment:
lastRoute.onExit must not be null at this point due to the if condition

Reply:
fixed